git_wrapper/base.py

```diff
@@ -69,7 +71,11 @@ def describe(self, sha):
            :return dict: A dict with tag and patch data
         """
         ret_data = {'tag': '', 'patch': ''}
-        output = self.git.describe(sha).split('-g')
+        try:
+            output = self.git.describe(sha).split('-g')
+        except git.CommandError as ex:
+            raise exceptions.ReferenceNotFoundException("Could not find sha %s." % sha) from ex
```

Comment:
This is clearly a situation where we want to raise an exception.
I'd be more comfortable if we added the --tags or --all options as it will always fail without annotated tags in repo.

Reply:
I don't have an issue with adding the '--all' option here to handle annotated tags.